code/modules/mining/equipment/resonator.dm

```diff
@@ -114,3 +114,10 @@
 	. = ..()
 	transform = matrix()*1.5
 	animate(src, transform = matrix()*0.1, alpha = 50, time = 4)
+
+/obj/effect/resonance/proc/replicate(turf/M, creator, timetoburst) 
+	for(var/turf/closed/T in orange(1, M))
+		if(istype(T, M))
```

Comment:
Why are you not just straight up typecasting for closed/mineral in orange loop?

You can change this to `istype(T)` after for very edge cases (admin spawns in mineral subtype)